$NetBSD: patch-ab,v 1.9 2009/07/27 12:59:44 obache Exp $

Fix out of bounds access in event_wallops().

This fixes irssi bug #662 (also known as CVE-2009-1959).

http://svn.irssi.org/cgi-bin/viewvc.cgi?view=rev&root=irssi&revision=5068

--- src/fe-common/irc/fe-events.c.orig	2009-07-15 22:54:17.000000000 +0200
+++ src/fe-common/irc/fe-events.c
@@ -298,7 +298,7 @@ static void event_wallops(IRC_SERVER_REC
 
 		tmp = g_strdup(data+8);
 		len = strlen(tmp);
-		if (tmp[len-1] == 1) tmp[len-1] = '\0';
+		if (len >= 1 && tmp[len-1] == 1) tmp[len-1] = '\0';
 		printformat(server, NULL, MSGLEVEL_WALLOPS, IRCTXT_ACTION_WALLOPS, nick, tmp);
 		g_free(tmp);
 	}
